docs/api/standard_library_types.md

```diff
@@ -151,3 +151,63 @@ m = Model(td='P3DT12H30M5S')
 print(m.model_dump())
 #> {'td': datetime.timedelta(days=3, seconds=45005)}
 ```
+
+## Number Types
+
+Pydantic supports the following numeric types from the Python standard library:
+
+### `int`
+
+* Pydantic uses `int(v)` to coerce types to an `int`;
+  see [Data conversion](../usage/models.md#data-conversion) for details on loss of information during data conversion.
+
+### `float`
+
+* Pydantic uses `float(v)` is used to coerce values to floats.
+
+### `enum.IntEnum`
+
+* Validation: Pydantic checks that the value is a valid `IntEnum` instance.
+
+#### subclass of `enum.IntEnum`
+
+* Validation: checks that the value is a valid member of the integer enum;
+  see [Enums and Choices](../usage/types/enums.md) for more details.
+
+
+### `decimal.Decimal`
+* Validation: Pydantic attempts to convert the value to a string, then passes the string to `Decimal(v)`.
+* Serialization: Pydantic serializes `Decimal` types as strings.
+You can use a custom serializer to override this behavior if desired. For example:
+
+```py
+from decimal import Decimal
+
+from typing_extensions import Annotated
+
+from pydantic import BaseModel, PlainSerializer
+
+
+class Model(BaseModel):
+    x: Decimal
+    y: Annotated[
+        Decimal,
+        PlainSerializer(
+            lambda x: float(x), return_type=float, when_used='json'
+        ),
+    ]
+
+
+my_model = Model(x=Decimal('1.1'), y=Decimal('2.1'))
+
+print(my_model.model_dump())  # (1)!
+#> {'x': Decimal('1.1'), 'y': Decimal('2.1')}
+print(my_model.model_dump(mode='json'))  # (2)!
+#> {'x': '1.1', 'y': 2.1}
+print(my_model.model_dump_json())  # (3)!
+#> {"x":"1.1","y":2.1}
+```
+
+1. Using `model_dump`, both `x` and `y` remain instances of the `Decimal` type
+2. Using `model_dump` with `mode='json'`, `x` is serialized as a `string`, and `y` is serialized as a `float` because of the custom serializer applied.
+3. Using `model_dump_json'`, `x` is serialized as a `string`, and `y` is serialized as a `float` because of the custom serializer applied.
```

Comment:
I'm not going to be able to comment this every time, because I also forget... But every mention to code can be linked to the API reference.

```suggestion
1. Using [`model_dump`][pydantic.main.model_dump], both `x` and `y` remain instances of the `Decimal` type
2. Using [`model_dump`][pydantic.main.model_dump] with `mode='json'`, `x` is serialized as a `string`, and `y` is serialized as a `float` because of the custom serializer applied.
3. Using [`model_dump_json`][pydantic.main.model_dump_json], `x` is serialized as a `string`, and `y` is serialized as a `float` because of the custom serializer applied.
```